rules/prefer-ternary.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const isParenthesized = require('./utils/is-parenthesized');
+
+const create = context => {
+	function checkIfStatement(node) {
+		if (isSingleBlockStatement(node) &&
+		isSameType(node) &&
+			checkConsequentAndAlternateType(node)) {
+			context.report({
+				node,
+				message: 'This `if` statement can be replaced by a ternary operator.',
+				fix(fixer) {
+					return fixFunction(node, fixer);
+				}
+			});
+		}
+	}
+
+	function fixFunction(node, fixer) {
+		const sourceCode = context.getSourceCode();
+		let prefix = '';
+		const ifCondition = sourceCode.getText(node.test);
+		let left = '';
+		let right = '';
+		if (getNodeBody(node.consequent).type === 'ExpressionStatement') {
+			const expressionType = getNodeBody(node.consequent).expression.type;
+			if (expressionType === 'AssignmentExpression') {
+				prefix = sourceCode.getText(getNodeBody(node.consequent).expression.left) + ' = ';
+				left = getNodeBody(node.consequent).expression.right;
+				right = getNodeBody(node.alternate).expression.right;
+			} else {
```

Comment:
Same as https://github.com/sindresorhus/eslint-plugin-unicorn/pull/514/files#r392754687

Reply:
Same as above, it wouldn't reach this line unless it was the same type